de-lib/src/main/java/org/iplantc/de/fileViewers/client/views/PathListViewer.java

```diff
@@ -138,33 +138,48 @@ void handleDropStatus(Object o,
                               CancellableEvent cancellableEvent,
                               StatusProxy statusProxy) {
 
-            if(!hasCorrectData(o)){
+            if(!isNonEmptyCollection(o)){
                 cancellableEvent.setCancelled(true);
                 statusProxy.setStatus(false);
+                return;
             }
-            // TODO Check to see if any items are pathlists. If they are, prevent drop
-            Iterable<DiskResource> iterable = (Iterable<DiskResource>) o;
-            for(DiskResource dr : iterable){
-                InfoType infoType1 = InfoType.fromTypeString(dr.getInfoType());
-                if(InfoType.HT_ANALYSIS_PATH_LIST.equals(infoType1)){
-                    cancellableEvent.setCancelled(true);
-                    statusProxy.update((SafeHtml)appearance::preventPathListDrop);
-                    statusProxy.setStatus(false);
-
-                    return;
+
+            if (hasDiskResources(o)) {
+
+                // TODO Check to see if any items are pathlists. If they are, prevent drop
+                Iterable<DiskResource> iterable = (Iterable<DiskResource>)o;
+                for (DiskResource dr : iterable) {
+                    InfoType infoType1 = InfoType.fromTypeString(dr.getInfoType());
+                    if (InfoType.HT_ANALYSIS_PATH_LIST.equals(infoType1)) {
+                        cancellableEvent.setCancelled(true);
+                        statusProxy.update((SafeHtml)appearance::preventPathListDrop);
+                        statusProxy.setStatus(false);
+
+                        return;
+                    }
                 }
+                cancellableEvent.setCancelled(false);
+                statusProxy.setStatus(true);
+            } else {
+                cancellableEvent.setCancelled(false);
+                statusProxy.setStatus(true);
             }
-            cancellableEvent.setCancelled(false);
-            statusProxy.setStatus(true);
         }
 
-        boolean hasCorrectData(Object data){
-           boolean isCollection = data instanceof Collection<?>;
+        boolean hasCorrectData(Object data) {
+
+            return isNonEmptyCollection(data)
+                       && hasDiskResources(data);
+        }
+
+        boolean hasDiskResources (Object data) {
+            return ((Collection<?>) data).iterator().next() instanceof DiskResource;
+        }
+
+        boolean isNonEmptyCollection(Object data) {
+            boolean isCollection = data instanceof Collection<?>;
             boolean isEmpty = ((Collection<?>) data).isEmpty();
```

Comment:
Should this check be done only when `isCollection` is `true`? Otherwise, this can can cause a `ClassCastException`